parcels/collection/collectionnodes.py

```diff
@@ -0,0 +1,1782 @@
+        If indices is None -> Particle cannot be retrieved -> Assert-Error and return None
+        If index is None, return last item (-1);
+        If index < 0: return from 'end' of collection.
+        If index in 'indices' is out of bounds, throws and OutOfRangeException.
+        If Particles cannot be retrieved, returns None.
+
+        :arg index: a list or numpy.ndarray of indices of Nodes to be popped (i.e. retrieved and removed) from this collections
+        :returns a list of retrieved Nodes
+        """
+        super().pop_multi_by_indices(indices)
+        results = []
+        for index in indices:
+            results.append(self.pop_single_by_index(index))
+            self._ncount -= 1
+        return results
+
+    def pop_multi_by_IDs(self, ids):
+        """
+        Searches for Particles with the IDs registered in 'ids', removes the Particles from the Collection and returns the Particles (or: their ParticleAccessors).
+        If Particles cannot be retrieved (e.g. because the IDs are not available), returns None.
+
+        :arg id: a list or numpy.ndarray of64-bit (signed or unsigned) integer IDs of Nodes to be popped
+                 (i.e. retrieved and removed) from this collections
+        :returns a list of retrieved Nodes
+        """
+        super().pop_multi_by_IDs(ids)
+        results = []
+        for id in ids:
+            results.append(self.pop_single_by_ID(id))
+            self._ncount -= 1
+        return results
+
+    def _clear_deleted_(self):
+        """
+        This (protected) function physically removes particles from the collection whose status is set to 'DELETE'.
+        It is the logical finalisation method of physically deleting particles that have been marked for deletion and
+        that have not otherwise been recovered.
+        This methods in heavily dependent on the actual collection type and should be implemented very specific
+        to the actual data structure, to remove objects 'the fastest way possible'.
+        """
+        node = self.begin()
+        while node is not None:
+            # ==== we need to skip here deleted nodes that have been queued for deletion, but are still bound in memory ==== #
+            if not node.is_valid():
+                node = node.next
+                continue
+            next_node = node.next
+            if node.data.state == OperationCode.Delete:
+                node.unlink()
+                self._data.remove(node)
+            node = next_node
+        self._ncount = len(self._data)
+
+    def merge(self, other=None):
+        """
+        This function merge two strictly equally-structured ParticleCollections into one. This can be, for example,
+        quite handy to merge two particle subsets that - due to continuous removal - become too small to be effective.
+
+        TODO - RETHINK IF THAT IS STILL THE WAY TO GO:
```

Comment:
Has this been done already? Or does this statement need to stay in the docstring for future developers?